app/workers/asset_manager_create_whitehall_asset_worker.rb

```diff
@@ -33,4 +35,24 @@ def perform(file_path, legacy_url_path, draft = false, attachable_model_class =
     FileUtils.rm(file)
     FileUtils.rmdir(File.dirname(file))
   end
+
+private
+
+  def create_asset_manager_asset(model_id, asset_variant, response)
+    return unless model_id && asset_variant
+
+    response_id = get_asset_id(response)
+    save_asset_id_to_assets(model_id, asset_variant, response_id)
+  end
+
+  def get_asset_id(response)
+    attributes = response.to_hash
+    url = attributes["id"]
+    url[/\/assets\/(.*)/, 1]
+  end
+
+  def save_asset_id_to_assets(model_id, variant, asset_manager_id)
+    asset = Asset.new(asset_manager_id:, attachment_data_id: model_id, variant:)
+    asset.save!
```

Comment:
Also, if we had the `AttachmentData` object, this would be more idiomatic if implemented using the [auto-generated ActiveRecord association method](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#module-ActiveRecord::Associations::ClassMethods-label-Auto-generated+methods):

```ruby
attachment_data.create_asset!(asset_manager_id:, variant:)
```

this way it'll automatically be associated with that Attachment Data object.

Reply:
That would have been lovely indeed, if only for not passing a billion arguments to this worker (quite the eyesore). But one cannot simply pass an object (insert LOTR meme here).